synapse/rest/synapse/client/pick_username.py

```diff
@@ -73,11 +69,40 @@ async def _async_render_GET(self, request: Request):
         return 200, {"available": is_available}
 
 
-class SubmitResource(DirectServeHtmlResource):
+class AccountDetailsResource(DirectServeHtmlResource):
     def __init__(self, hs: "HomeServer"):
         super().__init__()
         self._sso_handler = hs.get_sso_handler()
 
+        def template_search_dirs():
+            if hs.config.sso.sso_template_dir:
+                yield hs.config.sso.sso_template_dir
+            yield hs.config.sso.default_template_dir
+
+        self._jinja_env = build_jinja_env(template_search_dirs(), hs.config)
+
+    async def _async_render_GET(self, request: Request) -> None:
+        try:
+            session_id = get_username_mapping_session_cookie_from_request(request)
+            session = self._sso_handler.get_mapping_session(session_id)
+        except SynapseError as e:
+            logger.warning("Error fetching session: %s", e)
+            self._sso_handler.render_error(request, "bad_session", e.msg, code=e.code)
+            return
+
+        idp_id = session.auth_provider_id
+        template_params = {
+            "idp": self._sso_handler.get_identity_providers()[idp_id],
+            "user_attributes": {
+                "display_name": session.display_name,
+                "emails": session.emails,
+            },
+        }
+
+        template = self._jinja_env.get_template("sso_auth_account_details.html")
```

Comment:
Most of our code seems to get the templates "early" so that errors pop up during start-up instead of when a user hits the page. (And they do it in the config code...) This is different...but I'm not sure it matters.

I don't love loading all the templates early and storing them, sure you only render them once, but it means you're storing them in memory always. 🤷

Reply:
sorry yes. I mentioned this in the PR description; wdyt?

Reply:
I think that I should read PR descriptions more closely. It sounds like we're in agreement. Sorry about that!